connector/gitlab/gitlab.go

```diff
@@ -0,0 +1,255 @@
+// Package gitlab provides authentication strategies using Gitlab.
+package gitlab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/coreos/dex/connector"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+)
+
+const (
+	scopeEmail = "user:email"
+	scopeOrgs  = "read:org"
+)
+
+// Config holds configuration options for gilab logins.
+type Config struct {
+	BaseURL      string `json:"baseURL"`
+	ClientID     string `json:"clientID"`
+	ClientSecret string `json:"clientSecret"`
+	RedirectURI  string `json:"redirectURI"`
+}
+
+type gitlabUser struct {
+	ID       int
+	Name     string
+	Username string
+	State    string
+	Email    string
+	IsAdmin  bool
+}
+
+type gitlabGroup struct {
+	ID   int
+	Name string
+	Path string
+}
+
+// Open returns a strategy for logging in through GitLab.
+func (c *Config) Open(logger logrus.FieldLogger) (connector.Connector, error) {
+	return &gitlabConnector{
+		baseURL:      c.BaseURL,
```

Comment:
would you like to provide a default value for baseURL that corresponds to the public gitlab API URL? This would be helpful if the user does not provide one via the config.